src/hamiltonian.jl

```diff
@@ -71,8 +74,7 @@ function neg_energy(
     r::T,
     θ::T
 ) where {T<:AbstractVector}
-    _r = [abs2(r[i]) * h.metric.M⁻¹[i] for i in 1:length(r)]
-    return -sum(_r) / 2
+    return -sum(abs2(r[i]) * h.metric.M⁻¹[i] for i in 1:length(r))/2
```

Comment:
```julia
julia> using BenchmarkTools

julia> r = randn(10^3); M = randn(10^3);

julia> sumrm(r,M) = -sum(abs2(r[i]) * M[i] for i in 1:length(r))/2
sumrm (generic function with 1 method)

julia> function sumrmloop(r, M)
           T = promote_type(eltype(r),eltype(M))
           out = zero(T)
           @fastmath @inbounds @simd ivdep for i in eachindex(r)
               out += abs2(r[i]) * M[i]
           end
           T(-0.5)out
       end
sumrmloop (generic function with 1 method)


julia> @btime sumrmloop($r,$M)
  76.547 ns (0 allocations: 0 bytes)
-18.618864262885186

julia> @btime sumrm($r,$M)
  890.894 ns (2 allocations: 64 bytes)
-18.61886426288515
```
The vectorized version should also generally be more accurate, because of the separate accumulators:
```julia
julia> exact = sumrm(big.(r),big.(M))
-18.61886426288519053795358020254984371861193761668740174819653119986900805067409

julia> sumrmloop(r,M) - exact
4.615084666580908148219191771601062401748196531199869008050674090303841271023858e-15

julia> sumrm(r,M) - exact
4.014222145458591744177540515050731240174819653119986900805067409030384127102386e-14
```